File: 백준/Bronze/6246. 풍선 놀이/풍선 놀이.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int N = scanner.nextInt();
        int Q = scanner.nextInt();
        int[] balloon = new int[N + 1];
        for (int q = 0; q < Q; q++) {
            int L = scanner.nextInt();
            int I = scanner.nextInt();
            for (int i = L; i <= N; i += I) {
                balloon[i] = 1;
            }
        }
        int count = 0;
        for (int i = 1; i <= N; i++) {
            if (balloon[i] == 0) {
                count++;
            }
        }
        System.out.println(count);
        scanner.close();
    }
}

File: 백준/Bronze/6246. 풍선 놀이/풍선 놀이.c
#include <stdio.h>

int main() {
    int N, Q;
    scanf("%d %d", &N, &Q);
    int balloon[N + 1];
    for (int i = 0; i <= N; i++) {
        balloon[i] = 0;
    }
    for (int q = 0; q < Q; q++) {
        int L, I;
        scanf("%d %d", &L, &I);
        for (int i = L; i <= N; i += I) {
            balloon[i] = 1;
        }
    }
    int count = 0;
    for (int i = 1; i <= N; i++) {
        if (balloon[i] == 0) {
            count++;
        }
    }
    printf("%d\n", count);
    return 0;
}